pkg/tenant/tenant_test.go

```diff
@@ -98,8 +100,10 @@ func (suite *tenantTestSuite) SetupSuite() {
 	// Implements reset methods to ensure the server state doesn't change
 	// between tests in Before/After.
 	go func() {
-		if err := suite.srv.Serve(); err != nil {
-			suite.T().Logf("error occurred during service: %s", err)
+		if err := suite.srv.Serve(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			// This is a bad enough error that we should panic, otherwise tests will
+			// fail and it will be hard to debug why the tests failed.
+			panic(err)
```

Comment:
Added this to help debug where the problem with the tenant tests was coming from.